### DISCUSSION TITLE
I click the "Copilot: log in to github" button in Pycharm,  but nothing happened

### DISCUSSION THREAD

**Discussion by lanlanabcd** [community/community#7201] [Copilot Conversations] [answered]
I click the "Copilot: log in to github" button in Pycharm,  but nothing happened as said in the copilot-docs.

**Comment by jansorg** [comment]
It should've displayed a progress dialog with "Retrieving GitHub Device Code" or similar. Are you perhaps using a custom proxy or a VPN with PyCharm?

**Reply by Fuzzy9876** [reply]
I've got the same problem without using a custom proxy or a VPN :(
It does display "Retrieving GitHub Device Code", but nothing happened after that.

**Reply by lanlanabcd** [reply]
The cause and solution of my problem is the same as somenothing. Thank you！

**Comment by Fuzzy9876** [comment, marked as answer]
It was solved! First, make sure your Pycharm can connect to `https://github.com` correctly.
Check `Settings > Appearance & Behavior> System Settings > Http Proxy` and test it.
You might have to use proxy.

**Reply by lanlanabcd** [reply]
This method solves my problem. Thank you very much!!!

**Reply by wwccx** [reply]
: ) Nice! Solved my problem

**Reply by growvv** [reply]
有用！感谢！！

**Reply by Zhangkuns** [reply]
Nice!! Thanks very much!!

**Reply by jtizdev** [reply]
hey, for some reason didn't work out for me

**Reply by adduke** [reply]
It hasn't worked for me either.

**Comment by postrational** [comment]
I'm facing the same issue since a recent upgrade of the Copilot plugin. 

Setting the proxy did not help. I can connect to https://github.com correctly, but still stuck on "Retrieving GitHub Device Code".

When not using proxy, I get a JSON RPC error instead.

**Reply by BastLast** [reply]
same issue here with phpstorm

**Reply by BastLast** [reply]
turns out when you connect on webstorm you are already connected on phpstorm... I'm stupid

**Reply by physicalattraction** [reply]
8 months later, same issue. It used to work before I had upgraded my plugin.

**Reply by hear0me** [reply]
I have successed by this process
1. close pycharm
2. open global option of your vpn
3. open pycharm
4. choose Settings > Appearance & Behavior> System Settings > Http Proxy
5. choose second choise, auto detect proxy setting and test it
6. go back to retry login

**Reply by loftitechnologies** [reply]
@hear0me thank you. This worked for me.

**Reply by RunYouCleverBoy** [reply]
Nope, not working. 
No VPN here. Plain simple connection. 
Still here it comes and goes (quite eliminating wrong settings). Began 2 days ago.

Intermittently fails on any SAML attempt to connect to GitHub. Proxy settings (No proxy) tested and okay. 
Actually, it also fails on GitHub pushes from any Jetbrains IDE. I think it's because I first registered with SAML/HTTP. 
Also - tried a token - no success. Timeout.

I use gitHub console from Chrome. No problems so it's not a DNS issue.
Gosh, I don't know! 🤷 Ran out of ideas.

**Reply by ehudcandivore** [reply]
happening to me as well... exactly as you said - no proxy, no vpn, connection test works, and started about 2-3 days ago...

**Reply by RunYouCleverBoy** [reply]
I tried to `ping api.github.com` and it shows timeout (not always, intermittently).

But as that happens, calling from a browser to https://api.github.com  shows a list of services (which means - reaches github).
`curl https://api.github.com` does not work, either

I suspect that some genius in a local provider decided that `api.github.com` is some unsafe hackers' site. 🤦 🤬 🤦 

But I can't be sure till I try VPN

**Reply by ehudcandivore** [reply]
seems like you're on to something... The same thing happens to me (api.github.com works on a browser but not ping/curl):
Pinging ap-blockpage.prod.bzq.securingsam.com [2a05:d014:c59:1a03:1eee:b7e:d88f:dcb9] with 32 bytes of data:
Request timed out.
I'll try a VPN and see (pretty sure it'll fix the issue).

**Reply by ehudcandivore** [reply]
Yeah, connecting through a VPN solved it. So it's an ISP issue in my case, someone's blocking github's api address.

**Reply by RunYouCleverBoy** [reply]
Exactly. Same here. ISP issue, solved by VPN.

#### Observations
* Even a cell hot spot won't help because in some countries, most of the traffic, including cell carriers', go through the same internet service provider.
* Of course I tried to use their support. Nada. Over their paygrade 👎 .
* Since there is nobody to assist, I guess the only solution would be some VPN.

#### So why does it happen intermittently?
I guess their security "feature" server, or their DNS have load sharing. Those nice, yet incompetent ISP fellas configured some instances to block your traffic, and some - not 🤦. So it's a matter of chance whether you fall on a "feature" or just clear, normal traffic.

So this is what happens:

![image](https://github.com/community/community/assets/14019958/03115a37-bca7-4bb5-8044-3b0691e7e177)


## Questions left
* What exactly is blocked?

**Reply by conghuyn** [reply]
Try disabling IPv6. I disabled IPv6 on my Mac, then restarted the IDE, and everything worked.

**Reply by RunYouCleverBoy** [reply]
Oh, solved a year ago. My ISP company finally got sober 😀 and finally realized that GitHub is not a dangerous fraud site threatening the human race👏.
The worse part of that story was that I just could not workaround it with a cellular hot spot because my cellular carrier traffic went through the same ISP (corporate deals etc 🤬🤦🏻‍♂️).

Eventually, they got a hold over themselves and started behaving. Since then, problem solved.

**Comment by neoneo40** [comment]
I have solution.
First pay(30 free trial) then again. login success

**Comment by cesarve77** [comment]
I found one posible solution.
In my case the plugin was out date, it was installed but I just need to restart the IDE

**Reply by Ojasv-Singh-Zeotap** [reply]
worked

**Comment by flopana** [comment]
Disabling IPv6 fixed it for me for some reason.
![image](https://github.com/community/community/assets/28308182/2c94febc-66fb-4c0a-bc1d-98654aeb3efe)

**Reply by maodijim** [reply]
The solution works, had the same problem except the dst ipv6 address is different, I'm pretty sure it was working a few weeks back.

**Reply by RunYouCleverBoy** [reply]
It happened to me on IPv4, too 😢

**Comment by Liz4v** [comment]
Hi, I'm having this problem with WSL only. When I open PyCharm using the JetBrains Gateway and click login to github nothing happens. I would like an option in which I visit the site manually and acquire and type the code manually please.

**Comment by nieware** [comment]
I had a similar issue in PHPStorm, turned out it was due to the GitHub plugin being deactivated - not sure if I deactivated it myself, but some error message would have been nice. Instead, clicking "log in to github" did nothing - in the background, an exception was thrown and logged, after googling the exception message I arrived at https://youtrack.jetbrains.com/issue/IDEA-357181/Github-Copilot-Login-link-not-working and found the solution. Just mentioning it in case it helps anyone.
